x/twap/api.go

```diff
@@ -56,6 +55,18 @@ func (k Keeper) GetArithmeticTwap(
 	return k.getTwap(ctx, poolId, baseAssetDenom, quoteAssetDenom, startTime, endTime, arithmeticStrategy)
 }
 
+func (k Keeper) GetGeometricTwap(
+	ctx sdk.Context,
+	poolId uint64,
+	baseAssetDenom string,
+	quoteAssetDenom string,
+	startTime time.Time,
+	endTime time.Time,
+) (sdk.Dec, error) {
+	geometricStrategy := &geometric{k}
```

Comment:
nit:

We can probably define `geometricStrategy`, and `arithmeticStrategy` on the keeper. Then, initialize them both in keeper constructor.

Then, simply, do below:
```go
return k.getTwap(ctx, poolId, baseAssetDenom, quoteAssetDenom, startTime, endTime, k.geometricStrategy)
```

Reply:
This seems still TBD

Reply:
fixed lmk what you think